lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -50,6 +50,9 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
         ):
             self._compute_block_score(student, descendant_key, course_structure, scores_client, submissions_scores)
 
+        # self.scores may hold outdated data, force it to refresh on next access
+        lazy.invalidate(self, 'scores')
```

Comment:
ah yeah, true. I'll wrap it in a `try-finally` to ensure that any updated scores are made available.